tests/api/api.spec.ts

```diff
@@ -1145,6 +1145,59 @@ describe("@dataform/api", () => {
         expect(e.message).to.equal("Compilation timed out");
       }
     });
+
+    it("dataform.json is valid", async () => {
+      const graphInvalidWarehouse = await compile({
+        projectDir: path.resolve("df/examples/dataform_json_checks/invalid_warehouse")
+      });
+      const graphInvalidWarehouseErrors = utils.validate(graphInvalidWarehouse);
+      expect(graphInvalidWarehouseErrors)
```

Comment:
i realise you're following the pattern here, but i think we can start cleaning this up, at least for new tests. it would be much cleaner if we could just do:

```
expect(await compile(...)).to.eql({ ... });
```

note that if you do this, you might get cryptic errors from chai, complaining that 'foo' is not equal to 'foo' - this usually happens because you're comparing an instance of `proto.Foo` against a `proto.IFoo`, in which case just wrap your expected value in `proto.Foo.create(...)`.

let me know if you run into issues, or if it's a pain feel free to ignore this comment!